### ISSUE TITLE
Grafana drive details map jumping back and forth

### ISSUE THREAD

**Issue by cwanja** [teslamate-org/teslamate#3719]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When viewing a drive details screen, while mousing along the drive stats the Grafana map (plugin?) rendering the drive on a map is jumping back and forth along the drive. 

### Expected Behavior

Smooth dot following the drive stats based upon the positions.

### Steps To Reproduce

Fair warning - **not** all of my drives seem impacted
1. Launch a drive details dashboard
2. Mouse through the drive stats and watch the dot on the map and see if it jumps

Google Maps links relevant to the positions ID showing that the position IDs are gradually going North and not back and forth
[position ID 3440585](https://www.google.com/maps/place/29°44'14.3%22N+95°27'30.8%22W/@29.737305,-95.4611283,17z/data=!3m1!4b1!4m4!3m3!8m2!3d29.737305!4d-95.458548?hl=en&entry=ttu)
[position ID 3440586](https://www.google.com/maps/place/29°44'21.1%22N+95°27'29.7%22W/@29.739198,-95.4608163,17z/data=!3m1!4b1!4m4!3m3!8m2!3d29.739198!4d-95.458236?entry=ttu)
[position ID 3440587](https://www.google.com/maps/place/29°44'35.6%22N+95°27'27.2%22W/@29.743225,-95.4601333,17z/data=!3m1!4b1!4m4!3m3!8m2!3d29.743225!4d-95.457553?entry=ttu)

### Relevant log output

```shell
From the positions table, this is ID, latitude, longitude and date. 
 3440585 | 29.737305 | -95.458548 | 2024-03-02 21:17:23.426
 3440586 | 29.739198 | -95.458236 | 2024-03-02 21:17:38.886
 3440587 | 29.743225 | -95.457553 | 2024-03-02 21:17:54.305
```

### Screenshots

Notice the red dot, starting at the bottom of the map.

https://github.com/teslamate-org/teslamate/assets/54645561/ce755c9d-52b4-483a-920c-14de48e90017

### Type of installation

Docker

### Version

1.28.3

**Comment by cwanja** [comment]
Really not sure _if_ there is a fix, but open to a discussion on what might be the problem. As mentioned, it does not happen to every drive. And in fact, even the example above, it does not happen for the **whole** drive. Just portions of the drive.

Interested in others thoughts.

**Comment by cwanja** [comment]
> This usually is due to the StreamingAPI being enabled. Disable this in settings and the jumping stops.

Why would some drives that use the StreamingAPI not have the issue? 

The data is perfectly fine, see the links using the longitude and latitude links.

**Comment by dyxyl** [comment]
It is interesting that moving the mouse moves the dot on the map even though the time in the drive stats stays the same.  That suggests there are more positions with the same time to the nearest second.

Are there any other rows in the positions table for that range of dates?  E.g. try this (edit car_id = 1 for the id if your car if different):
```
SELECT
  id,
  date,
  latitude,
  longitude,
  power,
  speed
FROM positions
WHERE 
  car_id = 1 AND 
  date >= '2024-03-02 21:17:23' and date < '2024-03-02 21:17:55'
ORDER BY 
  date ASC;
```

**Comment by cwanja** [comment]
> It is interesting that moving the mouse moves the dot on the map even though the time in the drive stats stays the same. That suggests there are more positions with the same time to the nearest second.
> 
> Are there any other rows in the positions table for that range of dates? E.g. try this (edit car_id = 1 for the id if your car if different):
> 
> ```
> SELECT
>   id,
>   date,
>   latitude,
>   longitude,
>   power,
>   speed
> FROM positions
> WHERE 
>   car_id = 1 AND 
>   date >= '2024-03-02 21:17:23' and date < '2024-03-02 21:17:55'
> ORDER BY 
>   date ASC;
> ```

Only the three IDs I previously provided. 
```

teslamate=# SELECT
  id,
  date,
  latitude,
  longitude,
  power,
  speed
FROM positions
WHERE 
  car_id = 1 AND 
  date >= '2024-03-02 21:17:23' and date < '2024-03-02 21:17:55'
ORDER BY 
  date ASC;
   id    |          date           | latitude  | longitude  | power | speed 
---------+-------------------------+-----------+------------+-------+-------
 3440585 | 2024-03-02 21:17:23.426 | 29.737305 | -95.458548 |    31 |    34
 3440586 | 2024-03-02 21:17:38.886 | 29.739198 | -95.458236 |   127 |    64
 3440587 | 2024-03-02 21:17:54.305 | 29.743225 | -95.457553 |    -1 |   124
(3 rows)

teslamate=# 
```

**Comment by dyxyl** [comment]
I found one of my drives that also shows the problem if I zoom in very close (the timestamps are within a second of each other rather than your more spaced-out timestamps).

That was with grafana 9.5.16.

I then updated to grafana 10.3.3 and the problem went away, so it is probably a bug in grafana.

**Comment by cwanja** [comment]
Thanks for the review and confirmation @dyxyl. 

@JakobLichterfeld, do we have a way to test new Grafana versions or is just all manual to confirm dashboards work? @dyxyl, are you seeing any issues in `10.3.3` of Grafana with TeslaMate dashboards?

**Comment by dyxyl** [comment]
I've not found any problems with 10.3.3 yet. The only difference I've noticed is the map using cross hairs for tracking the position instead of a blob.

**Comment by cwanja** [comment]
Can confirm Grafana v10.4.0 (03f502a94d) has resolved the issue via [release 1.28.4](https://github.com/teslamate-org/teslamate/releases/tag/v1.28.4). Thanks @JakobLichterfeld for the release and @dyxyl for the identified fix.
